File: QtRobotGUI/Actor.cpp
// -*- c++ -*-

#include "RobotGame.h"

#include "Actor.h"

Actor::Actor (RobotGame &_theGame, Vector2D _thePosition, Vector2D _theSpeed,
	      const QColor &_theColor)
    : theGame(_theGame), thePosition(_thePosition), theSpeed(_theSpeed), theColor(_theColor)
{
}

char*
Actor::updateWithBlock (char *block)
{
    thePosition = *(Vector2D*)block;
    block += sizeof(Vector2D);

    theSpeed = *(Vector2D*)block;
    block += sizeof(Vector2D);

    return block;
}

Vector2D
Actor::position (void)
{
    return thePosition;
}

Vector2D
Actor::speed (void)
{
    return theSpeed;
}

const QColor&
Actor::color (void)
{
    return theColor;
}

File: RobotGUI/Actor.m
// -*- objc -*-

#import "Actor.h"

@implementation Actor

- (id) initWithGame: (RobotGame*) theGame
	   position: (Vector2D) thePosition
	      speed: (Vector2D) theSpeed
	      color: (NSColor*) theColor
{
    [super init];

    game = theGame;
    position = thePosition;
    speed = theSpeed;
    color = [theColor retain];

    return self;
}

- (void) dealloc
{
    [color release];

    [super dealloc];
}

- (char*) updateWithBlock: (char*) block
{
    position = *(Vector2D*)block;
    block += sizeof(Vector2D);

    speed = *(Vector2D*)block;
    block += sizeof(Vector2D);

    return block;
}

- (void) drawInView: (PlayFieldView*) view
{
}

- (NSColor*) color
{
    return color;
}

@end
